### DISCUSSION TITLE
Data-Logger window. Table / Graphs. File record / transfer

### DISCUSSION THREAD

**Discussion by Salva-OA** [pybricks/support#1967] [Ideas]
Hi,

It would be nice to have a window like the mu editor and some code to represent in a **line-grath** values grabbed by a sensor, variables of the program or parameters of an actuator (motor, lights ...) for its analysis in "real-time".
Something like the experiment label on the Lego EV3 LabView software, although simpler.

Also, some way to **generate a file**, that could be later retrieved from the hub and opened in a spreadsheet of your computer. Well, maybe with the help of an algorithm to prevent not exceeding the memory capacity of the hub (city /boost perhaps).

In this **“console” window** could be also useful to stop/start your program or make direct calls to a function subroutine for instance to trigger a ### keyboard event that would generate new values and new graphs.

Thanks for your support
Salva.

**Comment by laurensvalk** [comment]
If you don't mind doing one manual copy step, you can do it like this:

![image](https://github.com/user-attachments/assets/7129587e-0c3b-4e39-a7d0-be59996aecc7)

This approach gives you all the freedom that a spreadsheet tool provides, and the data is streamed so that there is no RAM limitation.

**Reply by Salva-OA** [reply]
Thanks a lot for the tip

Well, it is not "real-time" aprouch, but can be also a nice and easy maner to evaluate the resoult of the experiment.
